File: mpi/basic.f90
program basic

use mpi_f08, only : mpi_init, mpi_finalize

implicit none (type, external)

print *, "going to init MPI"

call MPI_INIT()

print *, "MPI Init OK"

call MPI_FINALIZE()

print *, "MPI closed"

end program

File: mpi/basic.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char **argv)
{
   int ierr;

   printf("going to init MPI\n");

   ierr = MPI_Init(&argc,&argv);
   if(ierr != 0) {
      perror("could not init MPI\n");
      return 1;
   }
   printf("MPI Init OK\n");

   ierr = MPI_Finalize();
   if(ierr != 0) {
      perror("could not close MPI\n");
      return 1;
   }

   printf("MPI closed\n");

   return 0;
}
